bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/handler/WemoLightHandler.java

```diff
@@ -266,16 +250,21 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                         + "&lt;/CapabilityValue&gt;&lt;/DeviceStatus&gt;" + "</DeviceStatusList>"
                         + "</u:SetDeviceStatus>" + "</s:Body>" + "</s:Envelope>";
 
-                String wemoURL = getWemoURL();
+                URL descriptorURL = service.getDescriptorURL(this);
+                String wemoURL = getWemoURL(descriptorURL, "bridge");
 
                 if (wemoURL != null && capability != null && value != null) {
-                    String wemoCallResponse = wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
-                    if (wemoCallResponse != null) {
-                        if (capability.equals("10008")) {
-                            OnOffType binaryState = null;
-                            binaryState = value.equals("0") ? OnOffType.OFF : OnOffType.ON;
-                            updateState(CHANNEL_STATE, binaryState);
+                    if (wemoHttpCaller != null) {
+                        String wemoCallResponse = wemoHttpCaller.executeCall(wemoURL, soapHeader, content);
```

Comment:
You should cache `wemoHttpCaller` to a local variable and perform your logic on that instead. That should fix the null checker warnings.

Reply:
Changed in all handlers